### PR TITLE
Feature/add mailer layout

### PR DESCRIPTION
**Pull request by orbanbotond**
Kudos are going to: @distroid

I just fixed the failing tests... (https://github.com/hanami/mailer/pull/73)

Finished in 0.13846 seconds (files took 0.73283 seconds to load)
54 examples, 0 failures, 2 pending

Randomized with seed 25

### CODE REVIEW

**Comment by distroid** [comment]
@orbanbotond Great! Now failed only RuboCop :)

**Comment by orbanbotond** [comment]
@distroid fixed ;)

**Comment by distroid** [comment]
@orbanbotond not so fast :)

**Comment by orbanbotond** [comment]
finally @distroid 
@jodosha it is ready to be pulled.

File: lib/hanami/mailer.rb [hunk 1]
```diff
@@ -214,9 +215,24 @@ def initialize(locals = {})
     # @since 0.1.0
     # @api private
     def render(format)
+      layout_template = self.class.layouts(format)
+      if layout_template&.exist?
```
Comment (jodosha) [inline lib/hanami/mailer.rb]: If the layout is declared by the mailer, but the file is missing, this should crash.

File: lib/hanami/mailer.rb [hunk 2]
```diff
@@ -214,9 +215,24 @@ def initialize(locals = {})
...
+      end
+    end
+
+    def render_without_layout(format)
       self.class.templates(format).render(self, @locals)
     end
 
+    def render_with_layout(layout_template, format)
+      layout_template.render(self, @locals) do
+        self.class.templates(format).render(self, @locals)
```
Comment (jodosha) [inline lib/hanami/mailer.rb]: Please use `render_without_layout` here.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -2,5 +2,11 @@
 require "hanami/utils"
 require "hanami/devtools/unit"
 require "hanami/mailer"
+require "pry"
+require "pry-nav"
+
+Hanami::Mailer.configure do
```
Comment (jodosha) [inline spec/spec_helper.rb]: Can you please find an alternative to this? In `unstable` branch global configuration is gone. I will have hard times to port these tests. Thanks!

**Comment by jodosha** [comment]
@orbanbotond @distroid Are you still interested in this feature?

**Comment by heragu** [comment]
Will this feature be included any time soon?

**Comment by jodosha** [comment]
@heragu Apparently @orbanbotond will resume the work here soon. Thank you all. 😺

**Comment by jodosha** [comment]
Closing as a stale PR. @orbanbotond feel free to resume the work at any time.

**Pull request closed without merging**